extensions/sql-database-projects/src/test/dialogs/publishOptionsDialog.test.ts

```diff
@@ -4,19 +4,64 @@
  *--------------------------------------------------------------------------------------------*/
 
 import * as should from 'should';
+import * as path from 'path';
+import * as os from 'os';
+import * as vscode from 'vscode';
 import * as testUtils from '../../test/testContext';
+import * as TypeMoq from 'typemoq';
 
 import { PublishOptionsDialog } from '../../dialogs/publishOptionsDialog';
 import { PublishDatabaseDialog } from '../../dialogs/publishDatabaseDialog';
 import { Project } from '../../models/project';
+import { ProjectsController } from '../../controllers/projectController';
+import { createContext, mockDacFxOptionsResult, TestContext } from '../testContext';
+import { emptySqlDatabaseProjectTypeId } from '../../common/constants';
+
+let testContext: TestContext;
 
 describe('Publish Database Options Dialog', () => {
+	before(async function (): Promise<void> {
+		testContext = createContext();
+	});
+
 	it('Should open dialog successfully ', async function (): Promise<void> {
 		const publishDatabaseDialog = new PublishDatabaseDialog(new Project(''));
 		const optionsDialog = new PublishOptionsDialog(testUtils.getDeploymentOptions(), publishDatabaseDialog);
 		optionsDialog.openDialog();
+
+		// Verify the dialog should exists
 		should.notEqual(optionsDialog.dialog, undefined);
 	});
-});
 
+	it('Should deployment options gets initialized correctly with sample test project', async function (): Promise<void>{
+		// Create new sample test project
+		const projController = new ProjectsController(testContext.outputChannel);
+		const projFileDir = path.join(os.tmpdir(), `TestProject_${new Date().getTime()}`);
+		const projFilePath = await projController.createNewProject({
+			newProjName: 'TestProjectName',
+			folderUri: vscode.Uri.file(projFileDir),
+			projectTypeId: emptySqlDatabaseProjectTypeId,
+			projectGuid: 'BA5EBA11-C0DE-5EA7-ACED-BABB1E70A575',
+			sdkStyle: false
+		});
+		const project = new Project(projFilePath);
```

Comment:
should be able to reuse the `createTestProject()` in testUtils.cs
https://github.com/microsoft/azuredatastudio/blob/d6dab3dd188db67807349026265b5d5a1613a6e5/extensions/sql-database-projects/src/test/testUtils.ts#L36